src/components/ha-target-picker.ts

```diff
@@ -31,13 +36,19 @@ import "./device/ha-device-picker";
 import type { HaDevicePickerDeviceFilterFunc } from "./device/ha-device-picker";
 import "./entity/ha-entity-picker";
 import type { HaEntityPickerEntityFilterFunc } from "./entity/ha-entity-picker";
-import "./ha-area-picker";
+import "./ha-area-floor-picker";
 import "./ha-icon-button";
 import "./ha-input-helper-text";
 import "./ha-svg-icon";
+import { SubscribeMixin } from "../mixins/subscribe-mixin";
+import {
+  FloorRegistryEntry,
+  subscribeFloorRegistry,
+} from "../data/floor_registry";
+import { AreaRegistryEntry } from "../data/area_registry";
 
 @customElement("ha-target-picker")
-export class HaTargetPicker extends LitElement {
+export class HaTargetPicker extends SubscribeMixin(LitElement) {
```

Comment:
Should we add floor to the hass object to avoid this?

Reply:
I dont think it will be used enough for that, but if it does we might have to yes...